enumerate_methods.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# This custom Enumerable module replicates the work of
+# ... the default Enumerable methods
+
+module Enumerable
+  def my_each
+    for i in (0...length)
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    for i in (0...length)
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    result = []
+    each do |current|
+      result.push(current) if yield(current)
+    end
+    return result
+  end
+
+  def my_all?
+    each do |current|
+      result = yield(current)
+      return false if result.nil? || result == false
+    end
+    return true
+  end
+
+  def my_none?
+    each do |current|
+      result = yield(current)
+      return false if result.nil? || result
+    end
+    return true
+  end
+
+  def my_count
+    counter = 0
+    each do |current|
+      result = yield(current)
+      counter += 1 if result == true
+    end
+    return counter
+  end
+
+  def my_map(proc = nil)
+    each do |current|
+      self[i] = yield(current)
+      self[i] = proc.call(current) if proc
```

Comment:
Since odin project specification said to use proc if proc and block given you can run this first and then otherwise do the yield

Reply:
So I refactored (without proc), I am sure you can figure out that for yourself

```ruby
def my_map(proc = nil)
    result = []
    my_each do |current, i|
      result <<  yield(current)

    end
    result
  end

```

Something like that works, instead of manipulating self, also don't forget to use your own each!